core/__init__.py

```diff
@@ -164,6 +179,11 @@ class ExtendableObject(object):
                 moduleInstance = moduleClass(moduleName, modulePath)
                 # Attach the module-specific or the default render
                 moduleInstance.render = render.get(moduleName, render["default"])(parent=moduleInstance)
+                module_indexes = []
+                for index in indexes:
+                    if index["kind"] == moduleName:
+                        module_indexes.append(index)
```

Comment:
You should consider of doing this stuff already in `load_indexes_from_file`. It might return a dict which already contains all index definitions grouped by module, and would run with a better performance, as objects are shared between equal modules in the render.